### PR TITLE
nad pkg added Update func to Builder

### PR DESCRIPTION
**Pull request by kononovn**

### CODE REVIEW

File: pkg/nad/builder.go [hunk 1]
```diff
@@ -155,6 +155,26 @@ func (builder *Builder) Delete() error {
...
+		return builder, err
+	}
+
+	glog.V(100).Infof("Updating NetworkAttachmentDefinition %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+
+	builder.Definition.CreationTimestamp = metaV1.Time{}
+	builder.Definition.ResourceVersion = builder.Object.ResourceVersion
```
Comment (evgenLevin) [inline pkg/nad/builder.go:170]: Why do we want to keep resorceversion?

Comment (kononovn) [reply pkg/nad/builder.go:170]: Yes, I'm not sure but it looks like it's required

Comment (evgenLevin) [reply pkg/nad/builder.go:170]: We don't do that  in the node package
https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/nodes/node.go#L63

**Pull request merged**